geopandas/sindex.py

```diff
@@ -273,10 +273,11 @@ class PyGEOSSTRTreeIndex(pygeos.STRtree):
         def __init__(self, geometry):
             # for compatibility with old RTree implementation, store ids/indexes
             original_indexes = geometry.index
-            non_empty = ~geometry.values.is_empty
-            self._geometries = geometry[non_empty]
-            self.objects = self.ids = original_indexes[non_empty]
-            super().__init__(self._geometries.values.data)
+            geometry = geometry.values.data.copy()
+            geometry[pygeos.is_empty(geometry)] = None
+            # set empty geometries to None to mantain indexing
```

Comment:
This is done to avoid that segfault, right? I'd be happier if it was resolved on pygeos side. We can keep this here, but instead of `# set empty geometries to None to mantain indexing` we should say why is that, link to pygeos issue and mention that it can be removed once resolved in upstream.

Reply:
I agree, I just don't know how easy it will be to track down the conditions to generate that segfault. It even seems to depend on the Pandas version! I'll add a comment, but I think for now we should move forward here.

Reply:
> I think for now we should move forward here

Agree, I just wanted to make sure that we know what is happening in our code.

Reply:
I opened an issue in pygeos with the little info we have, and linked to it in a comment in our code.